.github/actions/detect-workflow-js/src/detect.ts

```diff
@@ -62,11 +62,11 @@ export async function detectWorkflowFromContext(
 
   let [repository, ref, workflow] = ["", "", ""];
 
-  // If this is a slsa-github-generator repository or fork, then look
-  // for the repo and head SHA from the pull_request event value.
+  // If this is a pull request, then look for the repo and head SHA from the
+  // pull_request event value.
   if (
-    workflowData.event === "pull_request" &&
-    workflowData.repository.name === "slsa-github-generator"
+    workflowData.event === "pull_request" ||
+    workflowData.event === "merge_group"
```

Comment:
do we need to support PR for other repositories? SLSA 1.0 specs only requires supporting a subset of events, none of which are PR triggers.

Reply:
The repository name isn't `slsa-framework/slsa-github-generator` for GHA runs on PRs. They run in the fork and the fork could be named anything like `my-slsa-generator-fork`.

The reason I removed this is that this forces contributors to name their fork `slsa-github-generator`. If they change the name the pre-submits would fail and the check doesn't really improve security.

Reply:
Right. Maybe checking the `full_name` would be better.
https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository

Reply:
Done.